### ISSUE TITLE
[data grid] a11y the toolbar should be outside of the role='grid' element

### ISSUE THREAD

**Issue by tj36** [mui/mui-x#8525]
Hello, I am having an accessibility issue and I'm having trouble debugging this. 

<img src="https://user-images.githubusercontent.com/130041473/230316774-873341cc-c1ef-49f9-af59-4e53ade24429.png" width="500">

Please let me know if there is a resolution for this. Thanks.

**Comment by alexfauquette** [comment]
Where can we see your data grid, and what is the tool you're using for this report?

Because with only the generic message in your screenshot, we will not be able to investigate

**Comment by tj36** [comment]
Hi,
Below is the code I have.

`<DataGrid columns={columns}
							rows={notifications.data?.data?.data?.result?.entries ?? []}
							rowCount={rowCountState}
							loading={notifications.isLoading}
							page={pageNumber}
							pageSize={pageSize}
							paginationMode="server"
							onPageChange={newPageNumber => setPageNumber(newPageNumber)}
							autoHeight
						/>`

Below is the grid I see on the UI
![image](https://user-images.githubusercontent.com/130041473/230438045-0c5d8e14-5ada-4764-836e-3643d9e12a4c.png)

I am using storybook to test the code.
Please let me know if you need some more details.

**Comment by alexfauquette** [comment]
From what I understand, the problem comes from the toolbar. The `role="grid"` should not includes the toolbar. It only expect row and row headers (no button or input)

**Comment by RPiAwesomeness** [comment]
Just another data point, but I've run into this using `@mui/x-data-grid-pro` version `5.17.20`

**Comment by emilefleming** [comment]
@alexfauquette were you able to find a workaround?

**Comment by romgrk** [comment]
We should add to #7902 then.

**Comment by cherniavskii** [comment]
While changing aria attributes would be a breaking change, we can do this under the experimental `ariaV7` flag, so that we can unblock people and prepare these changes in advance.
@romgrk What do you think?
I looked into it while investigating https://github.com/mui/mui-x/issues/9403 and I can submit a draft PR.

**Comment by romgrk** [comment]
Yeah, fixing this would probably be a breaking change. I was thinking we could fix this and do the DOM layout change required for point 1 of #9171 at the same time.

I'm not sure how you see this working in a backwards-compatible way but if you have it figured out feel free to open a PR right away.

**Comment by cherniavskii** [comment]
@romgrk I have opened https://github.com/mui/mui-x/pull/9496, let's discuss it there

**Comment by oliviertassinari** [comment]
This issue is reported by Lighthouse too e.g. https://pagespeed.web.dev/analysis/https-mui-com-x/2irxt3dly6?hl=en&form_factor=desktop

<img width="892" alt="Screenshot 2023-06-29 at 02 33 11" src="https://github.com/mui/mui-x/assets/3165635/bb95506f-24bc-4576-8aa3-f2cf6cdbbb73">

And

**Comment by cherniavskii** [comment]
> I'm not sure, it could be sold as a bug fix.

Perhaps, but people might use the role in their tests as we do:
https://github.com/mui/mui-x/blob/889af4224cf32a28947fd2b0ddda09cbb2314337/packages/grid/x-data-grid/src/tests/layout.DataGrid.test.tsx#L238
This might break things for them.
